firebase-app/src/iosMain/kotlin/dev/gitlive/firebase/firebase.kt

```diff
@@ -30,20 +30,32 @@ actual class FirebaseApp internal constructor(val ios: FIRApp) {
     actual val name: String
         get() = ios.name
     actual val options: FirebaseOptions
-        get() = ios.options.run { FirebaseOptions(bundleID, APIKey!!, databaseURL!!, trackingID, storageBucket, projectID) }
+        get() = ios.options.run { FirebaseOptions(bundleID, APIKey!!, databaseURL!!, trackingID, storageBucket, projectID, GCMSenderID) }
 }
 
 actual fun Firebase.apps(context: Any?) = FIRApp.allApps()
     .orEmpty()
     .values
     .map { FirebaseApp(it as FIRApp) }
 
-actual fun FirebaseOptions.initialize(context: Any?): FirebaseOptions {
-    return when (context) {
-        is String -> FIROptions(contentsOfFile = context)
-        else -> FIROptions.defaultOptions()
-    }.run {
-        FirebaseOptions(applicationId, apiKey, databaseUrl, gaTrackingId, storageBucket, projectId)
+actual class FirebaseOptions actual constructor(
+    actual val applicationId: String,
+    actual val apiKey: String,
+    actual val databaseUrl: String?,
+    actual val gaTrackingId: String?,
+    actual val storageBucket: String?,
+    actual val projectId: String?,
+    actual val gcmSenderId: String?
+) {
+    actual companion object {
+        actual fun withContext(context: Any): FirebaseOptions? {
+            return when (context) {
+                is String -> FIROptions(contentsOfFile = context)
+                else -> FIROptions.defaultOptions()
+            }?.run {
+                FirebaseOptions(googleAppID!!, APIKey!!, databaseURL, trackingID, storageBucket, projectID, GCMSenderID)
```

Comment:
in android the used `apiKey` is the one with "api" lowercase passed from the constructor, while here is `APIkey`. The same is for other field used in the creation of `FirebaseOptions`. Does iOS sdk store this values in global variables?

Reply:
This values goes from `FIROptions` (it's `this` object inside `run` block) and this names are different between Android and iOS, thanks Google

Reply:
oh okay